hs_event_s3/hsevent/main.py

```diff
@@ -0,0 +1,69 @@
+import redpanda_connect
+import asyncio
+import json
+import django
+import threading
+
+django.setup()
+# django imports can only happen after django is setup
+from hs_core.models import BaseResource
+from hs_core.hydroshare.resource import delete_resource_file
+from theme.models import UserProfile
+from hs_core.views.utils import link_s3_file_to_django
+from hs_file_types.utils import get_logical_file_type, set_logical_file_type
+
+
+def link_s3_files_to_resource(resource, fullpath):
+    try:
+        res_file = link_s3_file_to_django(resource, fullpath)
+        # Create required logical files as necessary
+        if resource.resource_type == "CompositeResource":
+            file_type = get_logical_file_type(res=resource, file_id=res_file.pk, fail_feedback=False)
+            if not res_file.has_logical_file and file_type is not None:
+                set_logical_file_type(res=resource, user=None, file_id=res_file.pk, fail_feedback=False)
+    except Exception as e:
+        print(f"Error syncing resource {resource.short_id}: {e}")
+
+
+def sync_resource(file_updated, key, resource_id, username):
```

Comment:
This function calls the appropriate django functions for adding/removing files from the data/contents directory.